src/bsoncxx/document/element.cpp

```diff
@@ -58,8 +61,10 @@ std::uint32_t element::keylen() const {
 
 bsoncxx::type element::type() const {
     if (_raw == nullptr) {
-        throw bsoncxx::exception{error_code::k_unset_element,
-                                 "cannot return the type of uninitialized element"};
+        throw bsoncxx::exception{
+            error_code::k_unset_element,
+            "cannot return the type of uninitialized element" +
+                std::string(_key ? " with key \"" + std::string(_key.value().data()) + "\"" : "")};
```

Comment:
I had to wrap the second part of the string in `std::string` in order to get the ternary operator to work.